support/Environments/wheeler_clang.sh

```diff
@@ -4,7 +4,61 @@
 # See LICENSE.txt for details.
 
 spectre_setup_modules() {
-    echo "All modules on Wheeler are provided by the system"
+    if [ -z ${SPECTRE_HOME} ]; then
+        echo "You must set SPECTRE_HOME to the cloned SpECTRE directory"
+        return 1
+    fi
+
+    local start_dir=`pwd`
+    dep_dir=`realpath $1`
+    if [ $# != 1 ]; then
+        echo "You must pass one argument to spectre_setup_modules, which"
+        echo "is the directory where you want the dependencies to be built."
+        return 1
+    fi
+    mkdir -p $dep_dir
+    cd $dep_dir
+    mkdir -p $dep_dir/modules
+
+    cd $dep_dir
+
+    if [ -f $dep_dir/arpack/lib64/libarpack.a ]; then
+        echo "arpack is already set up"
+    else
+        echo "Installing arpack..."
+        wget https://github.com/opencollab/arpack-ng/archive/refs/tags/3.8.0.tar.gz
+        tar -xzf 3.8.0.tar.gz
+        mv arpack-ng-3.8.0 arpack-build
+        cd $dep_dir/arpack-build
+        mkdir build
+        cd build
+        cmake -D CMAKE_BUILD_TYPE=Release \
+              -D CMAKE_C_COMPILER=gcc \
+              -D BUILD_SHARED_LIBS=OFF \
+              -D CMAKE_INSTALL_PREFIX=$dep_dir/arpack ..
+        make -j4
+        make install
+        cd $dep_dir
+        rm 3.8.0.tar.gz
+        rm -r arpack-build
+        echo "Installed arpack into $dep_dir/arpack"
+        cat >$dep_dir/modules/arpack <<EOF
+#%Module1.0
+prepend-path LIBRARY_PATH "$dep_dir/arpack/lib64"
+prepend-path LD_LIBRARY_PATH "$dep_dir/arpack/lib64"
+prepend-path CPATH "$dep_dir/arpack/include"
+prepend-path CMAKE_PREFIX_PATH "$dep_dir/arpack/"
+EOF
+    fi
+
+    cd $start_dir
+
+    printf "\n\nIMPORTANT!!!\nIn order to be able to use these modules you\n"
+    echo "must run:"
+    echo "  module use $dep_dir/modules"
+    echo "You will need to do this every time you compile SpECTRE, so you may"
+    echo "want to add it to your ~/.bashrc."
```

Comment:
This might be for another PR, but we probably want to change the message to say "..., so if you want this SpECTRE environment to be your primary environment, you may want to add it to your ~/.bashrc."  Same wording as in the file for expanse.

(This is because if you do add it to your ./bashrc, it wreaks havoc if you want to compile SPECTRE in one environment while also running SPECTRE jobs in a different environment).

Reply:
Sounds reasonable, but let's do this in another PR 👍